src/main/java/Parser.java

```diff
@@ -0,0 +1,293 @@
+            break;
+        case "bye":
+            str = "Bye!! Hope to see you soon";
+            break;
+        case "number":
+            str = "Number of tasks to do is " + list.getSize();
+            break;
+        default:
+            if (inputs.length != 2) {
+                return "Invalid number of arguments";
+            }
+            str = update(inputs[1], list);
+        }
+        Storage.save(list);
+        return str;
+    }
+
+    /**
+     * Removes the user requested task from the tasklist
+     * @param request A string containing the integer value of the index.
+     * @param list The tasklist object containing all the stored tasks.
+     */
+
+    public static String delete(String request, TaskList list) {
+        try {
+            int index = Integer.parseInt(request);
+            isNumeric(index, list);
+            String str = "The event has been removed as per your request: \n "
+                    + list.get(index);
+            list.delete(index);
+            str += "Now you have" + list.getSize() + " tasks to the list";
+            return str;
+        } catch (NumberFormatException e) {
+            return "You have not provided a valid number";
+        } catch (InvalidNumberException e) {
+            return "The number provided was greater or lesser than the number of items in the list";
+        }
+    }
+
+    /**
+     * Marks the task at the user requested index as done
+     * @param request A String containing the integer value of the index
+     * @param list The tasklist object containing all the stored tasks.
+     */
+
+    public static String update(String request, TaskList list) {
+        try {
+            System.out.println(request);
+            int index = Integer.parseInt(request);
+            isNumeric(index, list);
+            list.updateStatus(index);
+            String str = "Nice! I've marked this task as done: \n "
+                    + list.get(index);
+            str += "Now you have " + list.getSize() + " tasks in the list";
+            return str;
+        } catch (NumberFormatException e) {
+            return "You have not provided a valid number";
+        } catch (InvalidNumberException e) {
+            return "The number provided was greater or lesser than the "
+                    + "number of items in the list";
+        }
+    }
+
+    /**
+     * Checking if the index provided by the user is of the right size
+     * @param n The index provided by the user
+     * @param list The tasklist object containing all the stored tasks.
+     * @throws InvalidNumberException
+     */
+    public static void isNumeric(int n, TaskList list) throws InvalidNumberException {
+        if (list.getSize() < n || n <= 0) {
+            throw new InvalidNumberException("The number provided is bigger tha the list size");
+        }
+    }
+
+    /**
+     * Adds a deadline object with description and datetime into the tasklist provided
```

Comment:
Is it better if an empty line can be added to separate the description and the @param?